shared/src/commonTest/kotlin/com/example/tasky/auth/data/AuthDataSourceTest.kt

```diff
@@ -0,0 +1,92 @@
+package com.example.tasky.auth.data
+
+import assertk.assertThat
+import assertk.assertions.isEqualTo
+import com.example.tasky.auth.data.model.LoginBody
+import com.example.tasky.auth.data.model.LoginResponse
+import com.example.tasky.auth.data.model.RegisterBody
+import com.example.tasky.commom.mock.HttpManagerMock
+import com.example.tasky.common.domain.model.ResultWrapper
+import io.ktor.client.engine.mock.MockEngine
+import io.ktor.client.engine.mock.respond
+import io.ktor.http.HttpHeaders
+import io.ktor.http.HttpStatusCode
+import io.ktor.http.headersOf
+import io.ktor.utils.io.ByteReadChannel
+import kotlinx.coroutines.test.runTest
+import kotlin.test.BeforeTest
+import kotlin.test.Test
+
+class AuthDataSourceTest {
+    private lateinit var authDataSource: AuthDataSource
+
+    @BeforeTest
+    fun setUp() {
+        val mockEngine =
+            MockEngine { request ->
+                if (request.url.encodedPath.contains("login")) {
+                    respond(
```

Comment:
Technically, the only thing your tests are testing here is that your `AuthDataSource` correctly wraps response data in Result.Success and correctly puts the error type in Result.Error. The rest doesn't really test anything else from your production code.
And for that you have to write a whole lot of code here. I'd personally recommend to only make use of MockServer when the class that needs it really involves some logic. Imagine a repository that coordinates syncing and caching calls and uses an HttpClient for that. That's way more logic to test than for your simple data source here (which I wouldn't write tests for at all - they'll just cause maintenance effort without ever finding a bug)